packages/tax/src/models/tax-rate.ts

```diff
@@ -56,17 +62,20 @@ export default class TaxRate {
   @Property({ columnType: "bool", default: false })
   is_combinable = false
 
-  @Property({ columnType: "text" })
-  tax_region_id: string
-
   @ManyToOne(() => TaxRegion, {
     fieldName: "tax_region_id",
     index: taxRegionIdIndexName,
-    cascade: [Cascade.REMOVE, Cascade.PERSIST],
+    mapToPk: true,
+    cascade: [Cascade.REMOVE],
   })
+  tax_region_id: string
+
+  @ManyToOne({ entity: () => TaxRegion, persist: false })
   tax_region: TaxRegion
 
-  @OneToMany(() => TaxRateRule, (rule) => rule.tax_rate)
+  @OneToMany(() => TaxRateRule, (rule) => rule.tax_rate, {
```

Comment:
question: can we create a tax_rate from a rule? if not (which I expect), in that case, the relation needs some rework in the rule model

Reply:
Yep you are right - take a look at the latest changes. wdyt?

Reply:
I would say lets go with that as all the tests are green, though the cascade are set at the application level we might need to revisit some of the relation definition any way. I would say lets re have a look at the details later on am move forward 🚀